extensions/integration-tests/setEnvironmentVariables.js

```diff
@@ -0,0 +1,101 @@
+
+// Environment variable names
+const ENVAR_AZURE_SERVER = 'AZURE_SQL';
+const ENVAR_AZURE_SERVER_USERNAME = 'AZURE_SQL_USERNAME';
+const ENVAR_AZURE_SERVER_PASSWORD = 'AZURE_SQL_PWD';
+const ENVAR_BDC_SERVER = 'BDC_BACKEND_HOSTNAME';
+const ENVAR_BDC_SERVER_USERNAME = 'BDC_BACKEND_USERNAME';
+const ENVAR_BDC_SERVER_PASSWORD = 'BDC_BACKEND_PWD';
+const ENVAR_STANDALONE_SERVER = 'STANDALONE_SQL';
+const ENVAR_STANDALONE_SERVER_USERNAME = 'STANDALONE_SQL_USERNAME';
+const ENVAR_STANDALONE_SERVER_PASSWORD = 'STANDALONE_SQL_PWD';
+const ENVAR_PYTHON_INSTALL_PATH = 'PYTHON_TEST_PATH';
+
+const SecretEnVarMapping = {};
+SecretEnVarMapping[SECRET_AZURE_SERVER] = ENVAR_AZURE_SERVER;
+SecretEnVarMapping[SECRET_AZURE_SERVER_PASSWORD] = ENVAR_AZURE_SERVER_PASSWORD;
+SecretEnVarMapping[SECRET_AZURE_SERVER_USERNAME] = ENVAR_AZURE_SERVER_USERNAME;
+SecretEnVarMapping[SECRET_BDC_SERVER] = ENVAR_BDC_SERVER;
+SecretEnVarMapping[SECRET_BDC_SERVER_PASSWORD] = ENVAR_BDC_SERVER_PASSWORD;
+SecretEnVarMapping[SECRET_BDC_SERVER_USERNAME] = ENVAR_BDC_SERVER_USERNAME;
+SecretEnVarMapping[SECRET_STANDALONE_SERVER] = ENVAR_STANDALONE_SERVER;
+SecretEnVarMapping[SECRET_STANDALONE_SERVER_PASSWORD] = ENVAR_STANDALONE_SERVER_PASSWORD;
+SecretEnVarMapping[SECRET_STANDALONE_SERVER_USERNAME] = ENVAR_STANDALONE_SERVER_USERNAME;
+
+const EnVarNameValueMapping = {};
+
+const AkvSecrets = [
+	SECRET_AZURE_SERVER,
+	SECRET_AZURE_SERVER_PASSWORD,
+	SECRET_AZURE_SERVER_USERNAME,
+	SECRET_BDC_SERVER,
+	SECRET_BDC_SERVER_PASSWORD,
+	SECRET_BDC_SERVER_USERNAME,
+	SECRET_STANDALONE_SERVER,
+	SECRET_STANDALONE_SERVER_PASSWORD,
+	SECRET_STANDALONE_SERVER_USERNAME
+];
+
+const EnvironmentVariables = [
+	ENVAR_AZURE_SERVER,
+	ENVAR_AZURE_SERVER_PASSWORD,
+	ENVAR_AZURE_SERVER_USERNAME,
+	ENVAR_BDC_SERVER,
+	ENVAR_BDC_SERVER_PASSWORD,
+	ENVAR_BDC_SERVER_USERNAME,
+	ENVAR_STANDALONE_SERVER,
+	ENVAR_STANDALONE_SERVER_PASSWORD,
+	ENVAR_STANDALONE_SERVER_USERNAME,
+	ENVAR_PYTHON_INSTALL_PATH
+];
+
+// Set the values that are not stored in AKV here
+EnVarNameValueMapping[ENVAR_PYTHON_INSTALL_PATH] = path.join(os.homedir(), 'azuredatastudio-python');
+
+let promises = [];
+
+// Fetch the values from AKV
+msrestAzure.interactiveLogin().then((credentials) => {
+	const client = new KeyVault.KeyVaultClient(credentials);
+	AkvSecrets.forEach(secret => {
+		let promise = client.getSecret(AKV_URL, secret, '').then((result) => {
+			console.log(`${secret}: ${result.value}`);
+			EnVarNameValueMapping[SecretEnVarMapping[secret]] = result.value;
+		}, (err) => {
+			console.error('An error occured while retrieving the key value');
+			console.error(err);
+		});
+		promises.push(promise);
+	});
+
+	Promise.all(promises).then(
+		() => {
+			let content = '';
+			EnvironmentVariables.forEach(envar => {
+				content += `${envar}=${EnVarNameValueMapping[envar]}${os.EOL}`;
+			});
+			fs.writeFileSync(path.join(os.homedir(), '.ads-int-test-env'), content);
```

Comment:
So it's ok to write an unencrypted file with the passwords to disk? If the security people say this is fine then cool, but it's just surprising that this is somehow a better option than passing the vars on the command line.

Reply:
I've thought some more about this, and I agree with Charles here. Can we ping AKV and just keep those variables in memory without writing to disk? Having unencrypted "secrets" in a local file scares me a bit.

Reply:
no more file involved here, resolving the chat thread